rustls/src/server/hs.rs

```diff
@@ -403,6 +416,17 @@ impl State for ExpectClientHello {
             ));
         }
 
+        if version == ProtocolVersion::TLSv1_3
+            && !self.done_retry
+            && client_hello.has_cookie_extension()
+        {
+            // RFC 8446 - 4.2.2 Cookie
+            // Clients MUST NOT use cookies in their initial ClientHello in subsequent connections.
```

Comment:
Same. Instead of citing/quoting the part of the spec that talks about what the client does, we should cite the part of the spec that requires us to fail with an error here.